frontend/src/routes/manage/Video/Single.tsx

```diff
@@ -0,0 +1,191 @@
+import { NotFound } from "../../NotFound";
+
+
+export const ManageSingleVideoRoute = makeRoute<PreloadedQuery<SingleVideoManageQuery>>({
+    path: "/~manage/videos/([a-zA-Z0-9\\-_]+)",
+    queryParams: [],
+    prepare: ({ pathParams: [videoId] }) => loadQuery(query, { id: `ev${videoId}` }),
+    render: queryRef => <QueryLoader {...{ query, queryRef }} render={result => (
+        result.event === null
+            ? <NotFound kind="video" />
+            : (
+                <Root nav={<BackLink />} userQuery={result}>
+                    {result.event.canWrite
+                        ? <ManageSingleVideo event={result.event}/>
+                        : <NotAuthorized />
+                    }
+                </Root>
+            )
+    )} />,
+});
+
+const BackLink: React.FC = () => {
+    const { t } = useTranslation();
+
+    const items = [
+        <LinkWithIcon key={ManageVideosRoute.path} to={ManageVideosRoute.path} iconPos="left">
+            <FiArrowLeft />
+            {t("manage.nav.my-videos")}
+        </LinkWithIcon>,
+    ];
+
+    return <LinkList items={items} />;
+};
+
+const query = graphql`
+    query SingleVideoManageQuery($id: ID!) {
+        ...UserData
+        event(id: $id) {
+            title
+            description
+            opencastId
+            thumbnail
+            created
+            updated
+            duration
+            canWrite
+            series { title, ...SeriesBlockSeriesData }
+            tracks { flavor resolution }
+        }
+    }
+`;
+
+type Event = NonNullable<SingleVideoManageQueryResponse["event"]>;
+
+type Props = {
+    event: Event;
+};
+
+const BREAKPOINT = 1100;
+
+const ManageSingleVideo: React.FC<Props> = ({ event }) => {
+    const { t } = useTranslation();
+
+    return <>
+        <h1>{t("manage.my-videos.video-details")}</h1>
+        <section css={{
+            width: 1100,
+            maxWidth: "100%",
+            marginBottom: 32,
+
+            [`@media(min-width: ${BREAKPOINT}px)`]: {
+                display: "flex",
+                flexDirection: "row-reverse",
+                gap: 48,
+                "& > *:first-child": {
+
+                },
+            },
+        }}>
+            <ThumbnailDateInfo event={event} />
+            <MetadataSection event={event} />
+        </section>
+        <section>
+            <TechnicalDetails event={event} />
+        </section>
+    </>;
+};
+
+const ThumbnailDateInfo: React.FC<Props> = ({ event }) => {
+    const { t, i18n } = useTranslation();
+    const created = new Date(event.created).toLocaleString(i18n.language);
+    const updated = new Date(event.updated).toLocaleString(i18n.language);
+
+    const DateValue = useCallback(({ label, value }) => (
+        <div css={{ "&:not(:last-child)": { marginBottom: 12 } }}>
+            <div css={{ color: "var(--grey40)", lineHeight: 1 }}>{label + ":"}</div>
+            <div css={{ marginLeft: 6, marginTop: 4 }}>{value}</div>
+        </div>
+    ), []);
```

Comment:
While this works, this seems weird to me. `useCallback` is not intended to define components ... :thinking:

Reply:
How would you do it then? I would like to have the component inside the other function (as it is only used there), but I don't want to redefine it every time the outer component is rendered. I mean the first requirement is not that hard...

Reply:
I would define it on the top level, I think. :thinking: